app/crud/async_db/vector_tiles/filters.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from fastapi import HTTPException
+from shapely.geometry import box
+from sqlalchemy.sql.elements import TextClause
+
+from ....application import db
+from ....utils.geostore import get_geostore_geometry
+
+Filter = Tuple[TextClause, Dict[str, Any]]
+Bounds = Tuple[float, float, float, float]
+
+
+async def geometry_filter(
+    geostore_id: Optional[str], bounds: Bounds
+) -> Optional[TextClause]:
+    if isinstance(geostore_id, str):
+        geometry, envelope = await get_geostore_geometry(geostore_id)
+        if not envelope.intersects(box(*bounds)):
+            raise HTTPException(
+                status_code=404, detail="Tile does not intersect with geostore"
+            )
+
+        f = filter_intersects("t.geom", geometry)  # TODO avoid having to use t.
+        return f
+    return None
+
+
+def confidence_filter(high_confidence_only: Optional[bool]) -> Optional[TextClause]:
+    if high_confidence_only:
+        # TODO make sure fire data are normalized
+        return db.text("(confidence__cat = 'high' OR confidence__cat = 'h')")
+    return None
+
+
+def contextual_filter(**fields: Union[str, bool]) -> List[TextClause]:
+    filters: List[TextClause] = list()
+    for field, value in fields.items():
+        if value is not None:
+            f = filter_eq(field, value)
+            filters.append(f)
+
+    return filters
+
+
+def date_filter(start_date: str, end_date: str) -> TextClause:
+    f: TextClause = db.text(
```

Comment:
should this accept the date field name too, since "alert__date" might not generically be the date field for other vector tables?

Reply:
right